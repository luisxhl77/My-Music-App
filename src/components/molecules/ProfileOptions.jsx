import { ArrowDropDown } from "@mui/icons-material"
import { Avatar } from "@mui/material"
import styled from "styled-components"
import { useSelector } from "react-redux"
import { selectUser } from "../../store/slices/UserSlice";
import { Link } from "react-router-dom";

export const ProfileOptions = () => {
  const user = useSelector(selectUser);
  const {display_name, images, id} = user.user;

  return (
    <ProfileContainer>
      <Avatar src={ images[0].url } alt={ id }/>
      <h4>{ display_name }</h4>
      <ArrowDropDown/>
      <div className="aquifue">
        <div>
        <Link to="/Profile">
          <button>perfil</button>
        </Link>
          <button>salir</button>
        </div>
      </div>
    </ProfileContainer>
  )
}

const ProfileContainer = styled.div`
  padding: 0px 0px;
  display: flex;
  align-items: center;
  color: #fff;
  border-radius: 30px;
  &:hover{
    background-color: #b3b3b3;
    cursor: pointer;
    & .aquifue{
      display: inherit;
    }
  }
  & h4{
    margin: auto 10px;
  }
  & .aquifue{
    display: flex;
    justify-content: center;
    z-index: -1;
    top: 0;
    right: 0;
    position: absolute;
    padding-top: 65px;
    height: 100px;
    width: 20vh;
    color: white;
    background-color: #181818;
    display: none;
    & div{
      background-color: red;
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;
      width: 100%;
    }
  }
`
