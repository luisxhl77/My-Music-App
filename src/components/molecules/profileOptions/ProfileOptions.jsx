import { NavLink, useNavigate } from "react-router-dom";
import PropTypes from 'prop-types';
import { ArrowDropDown } from "@mui/icons-material";
import { Avatar } from "@mui/material";
import './profileOptions.scss';

export const ProfileOptions = ({ name, image, id }) => {

  const navigate = useNavigate()
  
  const onLogout = () => {
    window.localStorage.clear();
    navigate('/Login', {
      replace: true,
    })
  }

  return (
    <div className="profileOptions">

      <NavLink to="/Profile" className={ ({ isActive }) => `profileOptions ${ isActive ? 'profileOptionsActive' : ''} `}>
        
        <Avatar src={ image } alt={ id } aria-label="imageUser"/>
        <h4 className="profileOptions__name" aria-label="nameUser"> { name } </h4>
        <ArrowDropDown/>
      
      </NavLink>
      
      <div className="profileOptions__options">
        
        <NavLink to="/Profile" className={ ({ isActive }) => `profileOptions__option ${ isActive ? 'activeoption' : ''} `}>
          <p> profile </p>
        </NavLink>
        
        <p className="profileOptions__option" onClick={onLogout} aria-label='funtionOnLogout'> onlogout </p>
      </div> 

    </div>
  )
}

ProfileOptions.propTypes = {
  name: PropTypes.string, 
  image: PropTypes.string, 
  id: PropTypes.string
}