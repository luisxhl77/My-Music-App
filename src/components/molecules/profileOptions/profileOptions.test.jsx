import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom";
import { ProfileOptions } from "./ProfileOptions"

describe('Prueba de <ProfileOptions/>', () => {
    const name  = 'Lucho Lopez';
    const image = 'https://avicii.com/avicii.jsg';
    const id    = 'asdasffeq';
    
    test('debe de hacer match con el snapshot' , () => {
        
        const { container } = render(
            <MemoryRouter>
                <ProfileOptions name={name} image={image} id={id}/> 
            </MemoryRouter>
        );
        expect( container ).toMatchSnapshot();
    })

    test('debe de mostrar el nombre del usuario' , () => {
        
        render(
            <MemoryRouter>
                <ProfileOptions name={name} image={image} id={id}/> 
            </MemoryRouter>
        );

        //investigar como mostrar los datos de un icono
        expect( screen.getByText( name )).toBe( screen.getByLabelText('nameUser') );
    })

    test('debe de ejecutar la funcion de onlogout' , () => {
        
        render(
            <MemoryRouter>
                <ProfileOptions name={name} image={image} id={id}/> 
            </MemoryRouter>
        );

        fireEvent.click( screen.getByLabelText('funtionOnLogout'));

    })

})