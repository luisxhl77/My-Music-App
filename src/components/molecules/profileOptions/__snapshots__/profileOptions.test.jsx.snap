// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <ProfileOptions/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="profileOptions"
  >
    <a
      class="profileOptions  "
      href="/Profile"
    >
      <div
        aria-label="imageUser"
        class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
      >
        <img
          alt="asdasffeq"
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="https://avicii.com/avicii.jsg"
        />
      </div>
      <h4
        aria-label="nameUser"
        class="profileOptions__name"
      >
         
        Lucho Lopez
         
      </h4>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m7 10 5 5 5-5z"
        />
      </svg>
    </a>
    <div
      class="profileOptions__options"
    >
      <a
        class="profileOptions__option  "
        href="/Profile"
      >
        <p>
           profile 
        </p>
      </a>
      <p
        aria-label="funtionOnLogout"
        class="profileOptions__option"
      >
         onlogout 
      </p>
    </div>
  </div>
</div>
`;
